### PR TITLE
fix: rehydration

### PR DESCRIPTION
**Pull request by ieow**
Fix rehydration using openlogin v6
fix error on unavailable currency rate

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

File: src/controllers/TorusController.ts [hunk 1]
```diff
@@ -1655,25 +1527,59 @@ export default class TorusController extends BaseController<TorusControllerConfi
...
-      publicKey: this.selectedAddress,
-      userInfo: req.params?.userInfo,
-      accounts: [
-        {
-          privKey: "",
-          address: this.selectedAddress,
-          solanaPrivKey: req.params?.privateKey,
-          app: `${req.params?.userInfo.email}`,
-          name: `${req.params?.userInfo.email}`,
+
```
Comment (ieow) [inline src/controllers/TorusController.ts:1536]: @chaitanyapotti , can you review this

File: src/auth/OpenLogin.ts [hunk 2]
```diff
@@ -37,5 +48,120 @@ class OpenLoginFactory {
...
+    const openLoginState = instance.state;
+    const { privKey, tKey, oAuthPrivateKey } = openLoginState;
+
+    if (!privKey) {
+      throw new Error("Login unsuccessful");
+    }
+
+    const userInfo = instance.getUserInfo();
+
+    const { sk: secretKey } = getED25519Key(privKey.padStart(64, "0"));
```
Comment (chaitanyapotti) [inline src/auth/OpenLogin.ts:67]: this can be simplified because instance has ed25519Key on it already

File: src/controllers/TorusController.ts [hunk 3]
```diff
@@ -1577,7 +1450,6 @@ export default class TorusController extends BaseController<TorusControllerConfi
   private async requestAccounts(req: JRPCRequest<unknown>): Promise<string[]> {
     // Try to restore from backend (restore privatekey)
     this.embedController.update({ loginInProgress: true });
-    await this.restoreFromBackend();
     this.embedController.update({ loginInProgress: false });
```
Comment (chaitanyapotti) [inline src/controllers/TorusController.ts:1453]: this needs to change...

File: src/controllers/TorusController.ts [hunk 4]
```diff
@@ -1655,25 +1527,59 @@ export default class TorusController extends BaseController<TorusControllerConfi
...
+    //     whiteLabel: state.whiteLabel || {},
+    //     loginConfig: {},
+    //   })
+    // )
+    const privateKey = req.params?.privateKey;
+
+    const openLoginHandler = await OpenLoginFactory.getInstance();
+    const { sessionId: openloginSessionId, state: openloginState } = openLoginHandler;
+    // this is import private key into torus wallet
+    if (openloginSessionId && openloginState.walletKey === privateKey) {
```
Comment (chaitanyapotti) [inline src/controllers/TorusController.ts:1551]: solana doesn't use walletKey.. it uses openloginState.privKey

File: src/controllers/TorusController.ts (hunk 3 above)
Comment (chaitanyapotti) [reply src/controllers/TorusController.ts:1453]: the flow needs to wait for openlogin session to resolve and then, respond here

Comment (ieow) [reply src/controllers/TorusController.ts:1484]: Added to resolve openlogin

**Pull request merged**